tests/client/core/components/TestInstallButton.js

```diff
@@ -1,195 +1,68 @@
-    const install = sinon.stub();
-    const uninstall = sinon.stub();
-    const button = renderButton({ status: DOWNLOADING, install, uninstall });
-    const root = findDOMNode(button);
-    Simulate.click(root);
-    assert.ok(!install.called);
-    assert.ok(!uninstall.called);
-  });
-
-  it('should associate the label and input with id and for attributes', () => {
-    const button = renderButton({ status: UNINSTALLED, slug: 'foo' });
-    const root = findDOMNode(button);
-    assert.equal(root.querySelector('input').getAttribute('id'),
-                'install-button-foo', 'id is set');
-    assert.equal(root.querySelector('label').getAttribute('for'),
-                'install-button-foo', 'for attribute matches id');
-  });
-
-  it('should throw on bogus status', () => {
-    assert.throws(() => {
-      renderButton({ status: 'BOGUS' });
-    }, Error, 'Invalid add-on status');
-  });
-
-  it('should not throw for ENABLING', () => {
-    renderButton({ status: ENABLING });
-  });
-
-  it('should not throw for DISABLING', () => {
-    renderButton({ status: DISABLING });
-  });
-
-  it('should call installTheme function on click when uninstalled theme', () => {
-    const installTheme = sinon.spy();
-    const browsertheme = { theme: 'data' };
-    sinon.stub(themePreview, 'getThemeData').returns(browsertheme);
-    const guid = 'test-guid';
-    const name = 'hai';
-    const button = renderButton({
-      installTheme,
-      type: THEME_TYPE,
-      guid,
-      name,
-      status: UNINSTALLED,
-    });
-    const root = findDOMNode(button.switchEl);
-    Simulate.click(root);
-    assert.ok(installTheme.calledOnce, installTheme.callCount);
-    const themeDataEl = installTheme.args[0][0];
-    assert.equal(themeDataEl.getAttribute('data-browsertheme'), JSON.stringify(browsertheme));
+  it('renders InstallSwitch when mozAddonManager is available', () => {
+    const i18n = getFakeI18nInst();
+    const root = shallowRender(<InstallButtonBase foo="foo" hasAddonManager i18n={i18n} />);
+    assert.equal(root.type, InstallSwitch);
+    assert.deepEqual(root.props, { foo: 'foo', hasAddonManager: true, i18n });
   });
 
-  it('should call install function on click when uninstalled', () => {
-    const guid = '@foo';
-    const name = 'hai';
-    const install = sinon.spy();
+  it('renders a theme button when mozAddonManager is not available', () => {
+    const installThemeAction = sinon.spy();
+    const i18n = getFakeI18nInst();
+    const addon = { type: THEME_TYPE, id: 'foo@personas.mozilla.org' };
+    const root = shallowRender(
+      <InstallButtonBase
+        addon={addon} foo="foo" hasAddonManager={false} i18n={i18n}
+        installThemeAction={installThemeAction} />);
+    assert.equal(root.type, Button);
+    assert.deepEqual(root.props, {
+      children: 'Install Theme',
+      'data-browsertheme': JSON.stringify(themePreview.getThemeData(addon)),
+      onClick: installThemeAction,
+    }, sinon.format(root.props));
```

Comment:
using `sinon.format()` is interesting! You reminded me that we need to solve this better though: https://github.com/mozilla/addons-frontend/issues/1447